runtime/lua/vim/diagnostic.lua

```diff
@@ -618,6 +594,7 @@ end
 ---                * priority: (number, default 10) Base priority to use for signs. When
 ---                {severity_sort} is used, the priority of a sign is adjusted based on
 ---                its severity. Otherwise, all signs use the same priority.
+---                * text: a table to set sign text
```

Comment:
Can you elaborate on what this configuration option does? I don't know this means.

Reply:
add sign text (icon) `sign = { text = { 'E', 'W', 'I' } } `

Reply:
I think it can be better described if you make it a table containing severity -> text mappings: `sign = { text = { ERROR = 'E', WARNING = 'W' ... } }`.

I think it should be considered a breaking change if the sign define API for these signs is removed. But perhaps one that is acceptable if Neovim wants to further encourage extmark usage over legacy signs?

Reply:
> But perhaps one that is acceptable if Neovim wants to further encourage extmark usage over legacy signs?

that's true. setting diagnostic text via sign define is just cumbersome. should be taken over by vim.diagnositc.config.